src/rgw/rgw_op.cc

```diff
@@ -4042,8 +4082,16 @@ void RGWPutObj::execute(optional_yield y)
   boost::optional<RGWPutObj_Compress> compressor;
 
   std::unique_ptr<rgw::sal::DataProcessor> encrypt;
+  std::unique_ptr<rgw::sal::DataProcessor> run_lua;
 
   if (!append) { // compression and encryption only apply to full object uploads
+    op_ret = get_lua_filter(&run_lua, filter);
+    if (op_ret < 0) {
+      return;
+    }
+    if (run_lua) {
+      filter = &*run_lua;
+    }
```

Comment:
don't we want to add this lua filter *last* so its data isn't transformed by compression or encryption? have you tested these combinations?

Reply:
> running lua on compressed and encrypted data won't be useful.
> this is why i added it first in the "put" case.

right, i just think you have it backwards. this code is adding wrapper filters around the object processor, so the filter added last is the one on the outside, so it sees the data first

Reply:
the "get" case looks correct as-is. there we're wrapping filters around `send_response_data()`. you're adding the lua filter before decompress/decrypt filters, so your lua filter should see the data in plain-text. it's still worth testing to confirm

Reply:
the "get" case was reading the compressed data.
i moved it to be first in 3ca1a4836906928323d6f4fa3ca1bd353a74c432
so it is executed last